packages/dds/tree/src/core/tree/detachedFieldIndex.ts

```diff
@@ -35,15 +37,43 @@ import { IIdCompressor } from "@fluidframework/id-compressor";
  */
 export type ForestRootId = Brand<number, "tree.ForestRootId">;
 
+/**
+ * fake revision used to mark that the revision stored in a {@link DetachedFieldIndex} is not yet
+ * set after loading data from a summary
+ */
+export const fakeRevisionWhenNotSet = Number.NaN as SessionSpaceCompressedId;
+
 /**
  * The tree index records detached field IDs and associates them with a change atom ID.
  */
 export class DetachedFieldIndex {
-	// TODO: don't store the field key in the index, it can be derived from the root ID
-	private detachedNodeToField: NestedMap<Major, Minor, ForestRootId> = new Map();
+	/**
+	 * A mapping from detached node ids to forest root ids and the revision that last created or modified the root.
+	 *
+	 * @remarks
+	 * undefined revisions are tolerated but any roots not associated with a revision must be disposed manually
+	 */
+	private detachedNodeToField: NestedMap<
+		Major,
+		Minor,
+		{ readonly root: ForestRootId; readonly latestRelevantRevision: RevisionTag | undefined }
+	> = new Map();
+	/**
+	 * A map between revisions and all roots for which the revision is the latest relevant revision.
```

Comment:
Would be good to explain here what it means for a revision to be the latest relevant revision (both in terms of conditions need to be met for a revision to be the latest relevant one, and in terms of what that means for GC)